examples/integration_test.py

```diff
@@ -402,6 +402,32 @@ def print_commit_result(err, partitions):
         print('# Committed offsets for: %s' % partitions)
 
 
+def verify_consumer_seek(c, seek_to_msg):
+    """ Seek to message and verify the next consumed message matches.
+        Must only be performed on an actively consuming consumer. """
+
+    tp = confluent_kafka.TopicPartition(seek_to_msg.topic(),
+                                        seek_to_msg.partition(),
+                                        seek_to_msg.offset())
+    print('seek: Seeking to %s' % tp)
+    c.seek(tp)
+
+    while True:
+        msg = c.poll()
+        assert msg is not None
+        if msg.error():
```

Comment:
comment is unrelated to this PR, but having non error events exposed via `error()` is not a good API choice imo. it'll confuse people.

Reply:
Right, the idea was to avoid exceptions in the fast path and treat consumer errors as non-fatals, which maps to what librdkafka is doing, but yes, it does confuse people and we haven't been great with documentation what errors to handle how.

As for error vs event; the only non-error event is PARTITION_EOF (which can be turned off by config (which should be the default, really))